### PR TITLE
Remove Handheld Controller

### LINKED ISSUE #1224: Remove the HandheldControllerBackend
### Description of the task

Once #1223 is complete, we will no longer need the `HandheldControllerBackend`. It's only purpose was to spoof camera data to prevent the robot's internal state from drifting and messing up the control. It can now be replaced with the `RadioBackend`.

### Acceptance criteria

- [x] Delete the HandheldControllerBackend
- [ ] Use the `RadioBackend` in the `handheld_controller_main.cpp` target

### Blocked By

#1223

### PR DESCRIPTION
**Pull request by kelvinkoon**
### Description

Removing the `HandheldControllerBackend` since the internal state drift was resolved. Moving the functionality to `RadioBackend`.

### Testing Done

Code builds. Tested by Mat and approved.

### Resolved Issues

Resolves #1224 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/backend/radio_backend.h [hunk 1]
```diff
@@ -11,6 +11,14 @@ class RadioBackend : public Backend
 
     RadioBackend();
 
+    /**
+     * Creates a new PrimitiveGenerator
+     *
+     * @param controller_input_config The config for the RadioBackend
+     */
+    RadioBackend(
```
Comment (MathewMacDougall) [inline src/software/backend/radio_backend.h]: The RadioBackend shouldn't need the `HandheldControllerInputConfig` so I don't think we need to add a constructor. The `HandheldControllerInputConfig` controls how the `handheld_controller` generates primitives, but by the time they reach the backend they should just be seen as "normal primitives".

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]

**Comment by EvanMorcom** [comment]
LGTM pending Mat's comment
:100: 🥇

**Review by EvanMorcom — DISMISSED** [review]

File: src/software/backend/radio_backend.h (hunk 1 above)
Comment (kelvinkoon) [reply src/software/backend/radio_backend.h]: In `handheld_control_main.cpp`, the original code passes in `controller_config` to `std::make_shared<HandheldControllerBackend>(controller_config)`. Without the constructor, how would we pass in `controller_config` into `RadioBackend`? Or is this already handled by `controller` and `primitive_generator` in the same `handheld_control_main.cpp` file?

**Review by jonathanlew — DISMISSED** [review]
LGTM pending Mat's comment and testing on a handheld controller

File: src/software/backend/radio_backend.h (hunk 1 above)
Comment (MathewMacDougall) [reply src/software/backend/radio_backend.h]: The `RadioBackend` doesn't need the `controller_config`, only the `HandheldControllerBackend` needed it.

**Review by akhilveeraghanta — COMMENTED** [review]
LGTM pending testing, nice yeetage of code


Yeetage
The proper term for the collective variety of possible yeets, AKA discarding an item at a high velocity.
"I caught my shorts and shirt on fire and had to yeet myself into the nearby lake"

"Yo that's some serious yeetage bro"

**Review by MathewMacDougall — COMMENTED** [review]
Changes LGTM, would be good to get a quick sanity check to make sure the robot's control still doesn't drift. @kelvinkoon if you can't get your controller working lmk and I should be able to use mine.

**Comment by kelvinkoon** [comment]
> Changes LGTM, would be good to get a quick sanity check to make sure the robot's control still doesn't drift. @kelvinkoon if you can't get your controller working lmk and I should be able to use mine.

I tried it this morning and still no luck. Would appreciate if you could try too if you have time.

**Review by MathewMacDougall — APPROVED** [review]
works for me :+1:

**Review by jonathanlew — APPROVED** [review]
:+1:

**Review by EvanMorcom — APPROVED** [review]
LGTM 🧇 :fire:

**Pull request merged**